vaadin-grid-flow-integration-tests/src/main/java/com/vaadin/flow/component/grid/it/ItemClickListenerPage.java

```diff
@@ -30,17 +33,39 @@ public ItemClickListenerPage() {
         Div dblClickMsg = new Div();
         dblClickMsg.setId("dblClickMsg");
 
+        Div columnClickMsg = new Div();
+        columnClickMsg.setId("columnClickMsg");
+
+        Div columnDblClickMsg = new Div();
+        columnDblClickMsg.setId("columnDblClickMsg");
+
         Grid<String> grid = new Grid<>();
         grid.setItems("foo", "bar");
-        grid.addColumn(item -> item).setHeader("Name");
-        grid.addComponentColumn(item -> new Checkbox(item));
+        grid.addColumn(item -> item).setHeader("Name").setKey("Name");
+        grid.addComponentColumn(Checkbox::new);
+
+        grid.addItemClickListener(event -> {
+            clickMsg.add(event.getItem());
+            columnClickMsg.add(event.getColumn().getKey());
+        });
 
-        grid.addItemClickListener(event -> clickMsg.add(event.getItem()));
+        grid.addItemDoubleClickListener(event -> {
+            dblClickMsg
+                    .setText(String.valueOf(event.getClientY()));
+            columnDblClickMsg.setText(event.getColumn().getKey());
```

Comment:
One more thing here: There's a reason why writing to the message used `add` instead of `setText`. This way it catches bugs where duplicate events are fired, because in that case e.g. `columnDblClickMsg` would become "NameName" instead of "Name", and the test would fail. And I do remember actually seeing this kind of duplicate-event bugs before. So please change these two from `setText` to `add`, to make the test more robust.